### DISCUSSION TITLE
Running tests: instructions are unclear

### DISCUSSION THREAD

**Discussion by KipCrossing** [wevm/viem#996] [Question]
Hey! I'm trying to run the tests locally. Context:

```
$ node -v
v18.16.0
$ pnpm -v
7.25.0
```

The [instructions](https://github.com/wagmi-dev/viem/blob/main/.github/CONTRIBUTING.md#running-the-test-suite) say to add into my env vars:

```
VITE_ANVIL_FORK_URL=https://rpc
VITE_ANVIL_BLOCK_TIME=1
VITE_ANVIL_BLOCK_NUMBER=16280770
VITE_NETWORK_TRANSPORT_MODE=http
```

Quote: *`ANVIL_FORK_URL` can be for any RPC service provider (e.g. Alchemy or Infura). Now you are ready to run the tests! In one terminal session, spin up Anvil using `pnpm anvil`. Next, in a different terminal session, you have the following options for running tests:*

- *pnpm test — runs tests in watch mode*

**Firstly**, I assume that `ANVIL_FORK_URL` is meant to be `VITE_ANVIL_FORK_URL` (please confirm this assumption).

**Secondly**, when running `pnpm anvil` it's `Listening on 127.0.0.1:8545`. Then when I run `pnpm test` in a *'different terminal session'* I get: *`Error: listen EADDRINUSE: address already in use :::8545`* (let me know if you want the full error)

**Thirdly**, I need more details around: *"can be for any RPC service provider (e.g. Alchemy or Infura)"* - I tried just 'pnpm test' with  `VITE_ANVIL_FORK_URL=https://sepolia.infura.io/v3/API_KEY` but got problems with block numbers:

```
Error: Anvil exited: thread 'main' panicked at 'Failed to get block for block number: 16280770
latest block number: 4045904'
```

Is this meant to be using a mainnet endpoint? If not, what testnet should I be using?

Or perhaps `VITE_ANVIL_BLOCK_NUMBER=16280770` is example only?

---

I'm happy to update the doc after this discussion. If the docs can be clear for me (who's new to this) they should be clear for almost everyone coming to this.

**Comment by KipCrossing** [comment]
**UPDATE:** it looks like it will run on the mainnet. As contributors coming into this for the first time, we have no idea what the point of setting `VITE_ANVIL_BLOCK_NUMBER=16280770` (and the rest) is. 

Also `npm anvil` in a different terminal session doesn't seem to be required anymore 

Some tests are failing though:

```
 ❯ src/actions/public/call.test.ts (24) 73668ms
   ❯ errors (9) 30337ms
     × gas too low 10020ms
     × gas fee is less than block base fee 10006ms
     × insufficient funds 10007ms
```

...

```
 ❯ src/actions/public/multicall.test.ts (19) 19814ms
   × stress 10445ms
   ```

...

```
  Snapshots  1 failed
 Test Files  4 failed | 227 passed | 3 skipped (234)
      Tests  7 failed | 1863 passed | 11 skipped | 3 todo (1884)
   Start at  12:00:59
   Duration  1316.12s
```

It looks like [failing tests are being merged into main](https://github.com/wagmi-dev/viem/actions/runs/5780792794/job/15664939790) which also make the process confusing and a high barrier to entry for contributors.

**Reply by jxom** [reply]
Some test cases can be a little brittle when running all cases on multiple threads. You can press `f` to rerun them and they should pass.

**Reply by KipCrossing** [reply]
I re-ran it multiple times, but I still get the same 3 errors:

   × offchain: aggregated 10024ms
   × stress 10465ms
   × gets transaction (legacy)

Expecting:

`"chainId": undefined`

Got:

`"chainId": 1`

Which makes sense as I set `VITE_ANVIL_FORK_URL` to the mainnet
